### PR TITLE
Added unittest bootstrap, and config test.

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
I've added a bootstrap which allows for loading from the test namespace.

Additionally I've added a test for the Config class, which isn't complete of the die statements that are in the class under test. This should be replaced by Exceptions, which allows for catching and exception handeling. Or not and just let the exception bubble up towards the user.

### CODE REVIEW

**Comment by DannyvdSluijs** [comment]
Klaus can we merge this pull request. I am willing to add additional test, however this PR needs to be done first.

**Comment by DannyvdSluijs** [comment]
Bump!

**Comment by DannyvdSluijs** [comment]
@garygreen could look into this PR as well ? I would love to add more testing but would need this PR merged first.

**Comment by garygreen** [comment]
@DannyvdSluijs thanks for this. The only thing I would say is I'm not too keen on adding another dependency just for mocking files (vfsStream). Might I suggest that we refactor the config class slightly like so:

1. Any `die` calls are replaced with exceptions `ConfigException` that is caught in silex and does a `die` (that way it can be caught in tests.)
2. Any file/dir checks abstract into a function on the `Config` class so they can be mocked in phpunit if required (use mocks instead of vfsStream).

That would remove the need for vfsStream and simplify the tests.

File: tests/GitList/Test/ConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ * Generated by PHPUnit_SkeletonGenerator on 2012-09-21 at 19:41:31.
+ */
+class ConfigTest extends \PHPUnit_Framework_TestCase {
+
+    /**
+     * @var Config
+     */
+    protected $object;
```
Comment (garygreen) [inline tests/GitList/Test/ConfigTest.php]: Can this be renamed to `config` as object is a bit generic.

**Comment by DannyvdSluijs** [comment]
@garygreen Seem's legit, I can take on these changes. You are correct that adding the vfsStream dependency is because the Config class is calling file systems methods.

How do you / @klaussilveira  envision this ? I would opt for an interface called IConfigReader with a single public method read(file  : string) : string[], this would allow for config to be stored in any format being read by a concrete reader class, the initial one being an IniReader.

```php
interface IConfigReader
{
    public function read($file);
}

class IniConfigReader implements IConfigReader
{
    public function read($file)
    {
        ....
    }
}
```

I'm guessing that the vfsStream dependancy can be dropped as well if we can create some files specific for unit testing. This way we can postpone this rather large change and write this down as nice to have for the rewrite ? Any suggestions ?

**Comment by klaussilveira** [comment]
Why should we reinvent the wheel of configuration management? Projects like [Configula](https://github.com/caseyamcl/Configula) and [Symfony Config](http://symfony.com/doc/current/components/config/index.html) already do this pretty well. 

However, why should an application like GitList offer different configuration strategies? It's useless. It adds unnecessary complexity, increases the maintainability cost and offer virtually no advantages. How many times do you change the format in which your application configuration is stored? Do we really want to add this kind of complexity to the codebase just because someone might want to transform their GitList config.ini to config.xml, because he likes the syntax better?

**Pull request closed without merging**